### PR TITLE
Bug fix 3.7/overfetch in filter

### PR DESCRIPTION
**Pull request by mchacki**
Reverts an unintended behavior change in Filter Executor in comparison to 3.6

It will now again fetch full batches of documents instead of taking into account the requested limit.

Backport of:
https://github.com/arangodb/arangodb/pull/12214

"Fixes"
https://arangodb.atlassian.net/browse/BTS-139

Jenkins:
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11065/

### CODE REVIEW

File: tests/js/server/aql/aql-profiler.js [hunk 1]
```diff
@@ -342,39 +342,19 @@ function ahuacatlProfilerTestSuite () {
 
     testFilterBlock2 : function () {
       const query = 'FOR i IN 1..@rows FILTER i % 13 != 0 RETURN i';
-      const genNodeList = (rows) => {
+      const genNodeList = (rows, batches) => {
         // This is an array 1..rows where true means it passes the filter
         const list = Array.from(Array(rows)).map((_, index ) => {
```
Comment (Dronplane) [inline tests/js/server/aql/aql-profiler.js:347]: Is this list still needed?

File: tests/js/server/aql/aql-profiler.js [hunk 2]
```diff
@@ -388,38 +368,19 @@ function ahuacatlProfilerTestSuite () {
 
     testFilterBlock3 : function () {
       const query = 'FOR i IN 1..@rows FILTER i % 13 == 0 RETURN i';
-      const genNodeList = (rows) => {
+      const genNodeList = (rows, batches) => {
         // This is an array 1..rows where true means it passes the filter
         const list = Array.from(Array(rows)).map((_, index ) => {
```
Comment (Dronplane) [inline tests/js/server/aql/aql-profiler.js:373]: Is this list still needed?

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11067/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**